update CrossRefSubmissionStatus checks to use .value for enum

devincowan opened the pull request:
Resolves #5952
Resolves #5926
<!--

Please read, and add your text at the bottom of this message.

Thanks for contributing code to HydroShare. In order to maintain code quality and expedite this process, please assist the development team by making sure the following is present in this pull request.

For more information, see https://docs.google.com/document/d/1dzxqlZW5fKNEyQSeKiSFq-SmS-VOPCva95XXkBjPExs

-->

### Pull Request Checklist: 
- [ ] Positive Test Case Written by Dev

<!-- Enter steps that a QA engineer, stakeholder, or user documentation writer would follow to test the positive or "successful" case of the functionality your code provides or fixes -->

- [ ] Automated Testing

<!-- Our Jenkins Instance is set up to automatically test every commit from a pull request. Code coverage must not decrease so new functionality or code paths added during a bug fix must have appropriate tests written. Every test must pass, including PEP8 code formatting tests. -->

- [ ] Sufficient User and Developer Documentation

<!-- Please email your positive test case lbrazil@cuahsi.org, who will make the decision regarding user documentation. -->

- [ ] Passing Jenkins Build

<!-- Our Jenkins Instance is set up to automatically test every commit from a pull request. Every test must pass, including PEP8 code formatting tests. -->

- [ ] Peer Code review and approval

<!-- This is the process by which a peer developer on the HydroShare team will read the changeset, provide feedback, and ultimately give a formal approval to the code before it passes PR status. -->

### Positive Test Case
1. [Enter positive test case here]

pkdash reviewed (approved):
I assume all other custom enum types use have been also been updated.

devincowan commented:
Thanks @pkdash good catch, looks like we need to also update the `RelationTypes`, `QuotaStatus`, as well as all the hs_access_control enums and the hs_file_types enums...

So let me work on that...

devincowan commented:
Okay @pkdash I checked the other enums and they all look good to me.
I also added a management command to fix the DOIs that are stuck pending

devincowan commented:
```bash
# python manage.py update_crossref_deposit_for_stuck_pending
Total published resources to process: 15
Updated DOI in metadata for resource cb1e4d52b44f4d9ab1187162456ba04c to https://doi.org/10.4211/hs.cb1e4d52b44f4d9ab1187162456ba04cpending
0. Attempting to deposit metadata to crossref for res id cb1e4d52b44f4d9ab1187162456ba04c    
Successfully deposited metadata to crossref for res id cb1e4d52b44f4d9ab1187162456ba04c
Updated DOI in metadata for resource 4a601ece58f140669b3fe548c70189e1 to https://doi.org/10.4211/hs.4a601ece58f140669b3fe548c70189e1pending

1. Attempting to deposit metadata to crossref for res id 4a601ece58f140669b3fe548c70189e1
Successfully deposited metadata to crossref for res id 4a601ece58f140669b3fe548c70189e1
Updated DOI in metadata for resource dfa02b83bbde4ae3888ffafeb4446a5b to https://doi.org/10.4211/hs.dfa02b83bbde4ae3888ffafeb4446a5bpending

2. Attempting to deposit metadata to crossref for res id dfa02b83bbde4ae3888ffafeb4446a5b
Successfully deposited metadata to crossref for res id dfa02b83bbde4ae3888ffafeb4446a5b
Updated DOI in metadata for resource b6788237717c41e0bcc69bcaa851694f to https://doi.org/10.4211/hs.b6788237717c41e0bcc69bcaa851694fpending

3. Attempting to deposit metadata to crossref for res id b6788237717c41e0bcc69bcaa851694f
Successfully deposited metadata to crossref for res id b6788237717c41e0bcc69bcaa851694f
Updated DOI in metadata for resource e51d9821c8d84682b642eb0818ac3137 to https://doi.org/10.4211/hs.e51d9821c8d84682b642eb0818ac3137pending

4. Attempting to deposit metadata to crossref for res id e51d9821c8d84682b642eb0818ac3137
Successfully deposited metadata to crossref for res id e51d9821c8d84682b642eb0818ac3137
Updated DOI in metadata for resource 4058253204f44d5680ccd7f7ae4b1fde to https://doi.org/10.4211/hs.4058253204f44d5680ccd7f7ae4b1fdepending

5. Attempting to deposit metadata to crossref for res id 4058253204f44d5680ccd7f7ae4b1fde
Successfully deposited metadata to crossref for res id 4058253204f44d5680ccd7f7ae4b1fde
Updated DOI in metadata for resource 6a0a63daaa4e4540b9fdef00975c55a7 to https://doi.org/10.4211/hs.6a0a63daaa4e4540b9fdef00975c55a7pending

6. Attempting to deposit metadata to crossref for res id 6a0a63daaa4e4540b9fdef00975c55a7
Successfully deposited metadata to crossref for res id 6a0a63daaa4e4540b9fdef00975c55a7
Updated DOI in metadata for resource 4b455c9ac1864e1ba7a3bc90e91e8e0e to https://doi.org/10.4211/hs.4b455c9ac1864e1ba7a3bc90e91e8e0epending

7. Attempting to deposit metadata to crossref for res id 4b455c9ac1864e1ba7a3bc90e91e8e0e
Successfully deposited metadata to crossref for res id 4b455c9ac1864e1ba7a3bc90e91e8e0e
Updated DOI in metadata for resource fd7af4fbf5444a2e828fa5e837da752c to https://doi.org/10.4211/hs.fd7af4fbf5444a2e828fa5e837da752cpending

8. Attempting to deposit metadata to crossref for res id fd7af4fbf5444a2e828fa5e837da752c  
Successfully deposited metadata to crossref for res id fd7af4fbf5444a2e828fa5e837da752c
Updated DOI in metadata for resource e5a4fefc9bfa428cb6f32fcf57d3554f to https://doi.org/10.4211/hs.e5a4fefc9bfa428cb6f32fcf57d3554fpending

9. Attempting to deposit metadata to crossref for res id e5a4fefc9bfa428cb6f32fcf57d3554f
Successfully deposited metadata to crossref for res id e5a4fefc9bfa428cb6f32fcf57d3554f
Updated DOI in metadata for resource 05a0490e971f491fa64c62cbde499a6a to https://doi.org/10.4211/hs.05a0490e971f491fa64c62cbde499a6apending

10. Attempting to deposit metadata to crossref for res id 05a0490e971f491fa64c62cbde499a6a
Successfully deposited metadata to crossref for res id 05a0490e971f491fa64c62cbde499a6a
Updated DOI in metadata for resource 9b2b865acc51481babc74599470a3bfc to https://doi.org/10.4211/hs.9b2b865acc51481babc74599470a3bfcpending

11. Attempting to deposit metadata to crossref for res id 9b2b865acc51481babc74599470a3bfc
Successfully deposited metadata to crossref for res id 9b2b865acc51481babc74599470a3bfc
Updated DOI in metadata for resource 84911fe3496f44eebbce512a6e8d0db3 to https://doi.org/10.4211/hs.84911fe3496f44eebbce512a6e8d0db3pending

12. Attempting to deposit metadata to crossref for res id 84911fe3496f44eebbce512a6e8d0db3
Successfully deposited metadata to crossref for res id 84911fe3496f44eebbce512a6e8d0db3
Updated DOI in metadata for resource fafd9b02ee3b4496830511342dab64a3 to https://doi.org/10.4211/hs.fafd9b02ee3b4496830511342dab64a3pending

13. Attempting to deposit metadata to crossref for res id fafd9b02ee3b4496830511342dab64a3
Successfully deposited metadata to crossref for res id fafd9b02ee3b4496830511342dab64a3
Updated DOI in metadata for resource 85ab212b18f543b0bd3c7b749cfeef66 to https://doi.org/10.4211/hs.85ab212b18f543b0bd3c7b749cfeef66pending
14. Attempting to deposit metadata to crossref for res id 85ab212b18f543b0bd3c7b749cfeef66
Successfully deposited metadata to crossref for res id 85ab212b18f543b0bd3c7b749cfeef66

Success
```
